### PR TITLE
Allow user to create an Event Hub namespace and/or Event Hub when creating EventHubTrigger

### PR DESCRIPTION
**Pull request by nturinski**
Partially fixes https://github.com/microsoft/vscode-azurefunctions/issues/3550

### CODE REVIEW

File: src/commands/addBinding/settingSteps/eventHub/EventHubListStep.ts [hunk 1]
```diff
@@ -3,30 +3,53 @@
...
+import { EventHubCreateStep } from './EventHubCreateStep';
 import { IEventHubWizardContext } from './IEventHubWizardContext';
 
 export class EventHubListStep extends AzureWizardPromptStep<IEventHubWizardContext> {
     public async prompt(context: IEventHubWizardContext): Promise<void> {
-        const namespaceName: string = nonNullProp(context, 'namespaceName');
-        const resourceGroupName: string = nonNullProp(context, 'resourceGroupName');
+        if (!context.eventHubsNamespace) {
+            // if there is no event hub namespace, then we can't create an event hub but
+            // getSubWizard only gets called if prompt does so we can't use shouldPrompt to stiffle this prompt
```
Comment (bwateratmsft) [inline src/commands/addBinding/settingSteps/eventHub/EventHubListStep.ts:19]: Nit
```suggestion
            // getSubWizard only gets called if prompt does so we can't use shouldPrompt to stifle this prompt
```

**Review by bwateratmsft — APPROVED** [review]

**Review by bwateratmsft — APPROVED** [review]

File: src/commands/addBinding/settingSteps/eventHub/EventHubListStep.ts (hunk 1 above)
Comment (alexweininger) [reply src/commands/addBinding/settingSteps/eventHub/EventHubListStep.ts]: stiffle reminds me of puffles from club penguin

File: src/commands/addBinding/settingSteps/eventHub/EventHubNameStep.ts [hunk 2]
```diff
@@ -11,6 +11,6 @@ export class EventHubNameStep extends StringPromptStep {
     public shouldPrompt(context: IEventHubWizardContext & IBindingWizardContext): boolean {
         // If the user decides to create a new app setting, `EventHubListStep` will take care of prompting
         // Otherwise, prompt to manually enter the name of the event hub using this step
-        return !context.namespaceName && !context.eventhubname;
+        return !context.eventHubsNamespace && !context.eventhubname;
```
Comment (MicroFish91) [inline src/commands/addBinding/settingSteps/eventHub/EventHubNameStep.ts:14]: Nit: Camel case?

`eventhubname` => `eventHubName`

File: src/commands/appSettings/connectionSettings/eventHubs/IEventHubsConnectionWizardContext.ts [hunk 3]
```diff
@@ -3,23 +3,17 @@
...
     azureWebJobsStorageType?: StorageConnectionTypeValues;
     eventHubsConnectionType?: EventHubsConnectionTypeValues;
 
-    // Netherite
-    newEventHubsNamespaceName?: string;
-    eventHubsNamespace?: EHNamespace;
-    newAuthRuleName?: string;
-    authRule?: AuthorizationRule;
-    newEventHubName?: string;
+    // Netherite uses all of the eventhub namespace settings in IEventHubWizardContext
```
Comment (MicroFish91) [inline src/commands/appSettings/connectionSettings/eventHubs/IEventHubsConnectionWizardContext.ts:18]: Probably okay, but I'll double check some `Netherite` scenarios to make sure this doesn't break.  We'll probably want CTI to pay extra attention to this flow when they do their test pass

More reason for me to get some tests in for Durable as well...

File: src/commands/appSettings/connectionSettings/eventHubs/IEventHubsConnectionWizardContext.ts [hunk 4]
```diff
@@ -3,23 +3,17 @@
...
-import type { AuthorizationRule, EHNamespace } from "@azure/arm-eventhub";
-import type { ResourceGroup } from "@azure/arm-resources";
+import type { ResourceManagementModels } from "@azure/arm-resources";
 import { ISubscriptionContext } from "@microsoft/vscode-azext-utils";
 import { EventHubsConnectionTypeValues, StorageConnectionTypeValues } from "../../../../constants";
 import { ISetConnectionSettingContext } from "../ISetConnectionSettingContext";
 
 export interface IEventHubsConnectionWizardContext extends ISetConnectionSettingContext, Partial<ISubscriptionContext> {
-    resourceGroup?: ResourceGroup;
+    resourceGroup?: ResourceManagementModels.ResourceGroup;
```
Comment (MicroFish91) [inline src/commands/appSettings/connectionSettings/eventHubs/IEventHubsConnectionWizardContext.ts:12]: Should the types be more like this?

image

File: src/commands/addBinding/settingSteps/LocalAppSettingListStep.ts [hunk 5]
```diff
@@ -24,7 +25,6 @@ import { CosmosDBListStep } from './cosmosDB/CosmosDBListStep';
 import { EventHubAuthRuleListStep } from './eventHub/EventHubAuthRuleListStep';
 import { EventHubConnectionCreateStep } from './eventHub/EventHubConnectionCreateStep';
 import { EventHubListStep } from './eventHub/EventHubListStep';
-import { EventHubNamespaceListStep } from './eventHub/EventHubNamespaceListStep';
```
Comment (MicroFish91) [inline src/commands/addBinding/settingSteps/LocalAppSettingListStep.ts:27]: Are you missing an import for:  `EventHubsNamespaceListStep`?

Comment (nturinski) [reply src/commands/addBinding/settingSteps/LocalAppSettingListStep.ts:27]: No idea why that got removed...

File: src/commands/addBinding/settingSteps/eventHub/EventHubNameStep.ts (hunk 2 above)
Comment (nturinski) [reply src/commands/addBinding/settingSteps/eventHub/EventHubNameStep.ts:14]: There's a note in IEventHubWizardContext

```
  /**
     * NOTE: The name of this variable should not change. It matches the name of the binding setting written to function.json
     */
    eventhubname?: string;
```

File: src/commands/appSettings/connectionSettings/eventHubs/IEventHubsConnectionWizardContext.ts (hunk 4 above)
Comment (nturinski) [reply src/commands/appSettings/connectionSettings/eventHubs/IEventHubsConnectionWizardContext.ts:12]: Do you mean the spaces? I'm not sure what you're trying to highlight here.
Or the fact that it has `resourceGroup` as a property?

**Review by alexweininger — APPROVED** [review]

File: src/commands/appSettings/connectionSettings/eventHubs/IEventHubsConnectionWizardContext.ts [hunk 6]
```diff
@@ -3,17 +3,17 @@
...
+export interface IEventHubsConnectionWizardContext extends IEventHubWizardContext, ISetConnectionSettingContext {
+    resourceGroup?: ResourceGroup;
 
     // Connection Types
     azureWebJobsStorageType?: StorageConnectionTypeValues;
     eventHubsConnectionType?: EventHubsConnectionTypeValues;
 
     // Netherite uses all of the eventhub namespace settings in IEventHubWizardContext
-}
+}``
```
Comment (alexweininger) [inline src/commands/appSettings/connectionSettings/eventHubs/IEventHubsConnectionWizardContext.ts]: oopsies

Comment (nturinski) [reply src/commands/appSettings/connectionSettings/eventHubs/IEventHubsConnectionWizardContext.ts]: Damn it. The theme I _was_ using had quotes be like dark gray on a dark gray background so they were impossible to see.

Though I got rid of them and pushed those changes. Not sure why it's still showing up here

image

**Review by alexweininger — APPROVED** [review]

**Pull request merged**